src/Authentication.Abstractions.Test/TestData/ArmResponseNoAzureCloud.json

```diff
@@ -0,0 +1,57 @@
+[
+  {
```

Comment:
I also got a report on the same thread that there are additional errors that prevent module load when customers use the  ```ARM_CLOUD_METADATA_URL``` environment variable to point at a metadata endpoint that does not have the 'AzureCloud' entry.  I believe we have a test using the environment variable for discovery, and we should ensure that it also convers this scenario.

Reply:
You are right. The discovery approach via default DNS, parameter, or environment variable are going into the same logic. It should be covered.